### PR TITLE
Restructure how data is stored and loaded, and make the site more 'atmospheric'.

### PR DESCRIPTION
**Pull request by SirHall**
1. Moved all the categories into their own files to make it easier to manage their contents.
2. Write in a little asynchronous loader that loads in the category files after the main page has finished loading to speed up page load times.
3. Added YouTube audio player that enhances the atmospheric 'feel' of the generator. :)

### CODE REVIEW

File: ideagenerator/generator.js [hunk 1]
```diff
@@ -5,18 +5,27 @@ let recently_used = [];
...
+	if(!parsed)
+		parseCategories();
+	let result;
+	do
+	{
+		reset();
+		let template = pickRandom('template');
+		//Temporary fix for 'result' sometimes being null
+		result = fillInTemplate(template);
+	}while(typeof result === 'undefined' || result.length === 0);
```
Comment (MrSkwiggs) [inline ideagenerator/generator.js]:
```suggestion
	} while(typeof result === 'undefined' || result.length === 0);
```

File: ideagenerator/generator.js [hunk 2]
```diff
@@ -5,18 +5,27 @@ let recently_used = [];
...
+let parsed = false;
+
 function generate() {
-	reset();
-	let template = pickRandom('template');
-	let result = fillInTemplate(template);
+	if(!parsed)
+		parseCategories();
+	let result;
+	do
```
Comment (MrSkwiggs) [inline ideagenerator/generator.js]:
```suggestion
	do {
```

File: ideagenerator/generator.js [hunk 3]
```diff
@@ -5,18 +5,27 @@ let recently_used = [];
...
+	{
```
Comment (MrSkwiggs) [inline ideagenerator/generator.js]:
```suggestion
```

**Review by MrSkwiggs — CHANGES_REQUESTED** [review]
Really nice, will be way easier to add to this :)

File: ideagenerator/generator.js (hunk 1 above)
Comment (SirHall) [reply ideagenerator/generator.js]: I had to turn my auto-formater for this project so there may have been some sections that I may have missed. Thanks!

File: ideagenerator/generator.js (hunk 2 above)
Comment (SirHall) [reply ideagenerator/generator.js]: I personally prefer K&R so I'm mostly used to that.

**Review by MrSkwiggs — APPROVED** [review]

**Comment by SebLague** [comment]
Thanks for the improvements everyone
EDIT: have since found some commonly occurring issues with generated outputs (group/character names missing). Will look into when I have time and reinstate the changes (or if you could try fix and resubmit that would be great too!)

**Pull request merged**